lightly/models/modules/masked_autoencoder_timm.py

```diff
@@ -0,0 +1,480 @@
+        return torch.cat((class_emb.unsqueeze(0), pos_embed), dim=1)
+
+    def images_to_tokens(
+        self, images: torch.Tensor, prepend_class_token: bool
+    ) -> torch.Tensor:
+        """Converts images into patch tokens.
+
+        Args:
+            images:
+                Tensor with shape (batch_size, channels, image_size, image_size).
+
+        Returns:
+            Tensor with shape (batch_size, sequence_length - 1, hidden_dim)
+            containing the patch tokens.
+        """
+        tokens = torch.Tensor(self.patch_embed(images))
+        if prepend_class_token:
+            tokens = utils.prepend_class_token(tokens, self.cls_token)
+        return tokens
+
+    def _initialize_weights(self) -> None:
+        # Initialize the patch embedding layer like a linear layer instead of conv
+        # layer.
+        w = self.patch_embed.proj.weight.data
+        torch.nn.init.xavier_uniform_(w.view([w.shape[0], -1]))
+
+        # Initialize the class token.
+        torch.nn.init.normal_(self.cls_token, std=0.02)
+
+        # initialize nn.Linear and nn.LayerNorm
+        self.apply(_init_weights)
+
+        _initialize_2d_sine_cosine_positional_embedding(self.pos_embed)
+
+
+class MAEDecoder(Module):
+    """Decoder for the Masked Autoencoder model [0].
+
+    Decodes encoded patches and predicts pixel values for every patch.
+    Code inspired by [1].
+
+    - [0]: Masked Autoencoder, 2021, https://arxiv.org/abs/2111.06377
+    - [1]: https://github.com/facebookresearch/mae
+
+    Attributes:
+        num_patches:
+            Number of patches.
+        patch_size:
+            Patch size.
+        in_chans:
+            Number of image input channels.
+        embed_dim:
+            Embedding dimension of the encoder.
+        decoder_embed_dim:
+            Embedding dimension of the decoder.
+        decoder_depth:
+            Depth of transformer.
+        decoder_num_heads:
+            Number of attention heads.
+        mlp_ratio:
+            Ratio of mlp hidden dim to embedding dim.
+        proj_drop_rate:
+            Percentage of elements set to zero after the MLP in the transformer.
+        attn_drop_rate:
+            Percentage of elements set to zero after the attention head.
+        norm_layer:
+            Normalization layer.
+
+    """
+
+    def __init__(
+        self,
+        num_patches: int,
+        patch_size: int,
+        in_chans: int = 3,
+        embed_dim: int = 1024,
+        decoder_embed_dim: int = 512,
+        decoder_depth: int = 8,
+        decoder_num_heads: int = 16,
+        mlp_ratio: float = 4.0,
+        proj_drop_rate: float = 0.0,
+        attn_drop_rate: float = 0.0,
+        norm_layer: Callable[..., nn.Module] = partial(LayerNorm, eps=1e-6),
+    ):
+        super().__init__()
+
+        self.decoder_embed = nn.Linear(embed_dim, decoder_embed_dim, bias=True)
+        self.mask_token = nn.Parameter(torch.zeros(1, 1, decoder_embed_dim))
```

Comment:
We have to decide where the mask token is stored. Right now it is here but there is also a mask token in `examples/pytorch/mae_timm.py` in the main module. From the example code I think the one here is never used (except in the initialization).